source/cppfs/source/FileEventHandler.cpp

```diff
@@ -0,0 +1,43 @@
+
+#include <cppfs/FileEventHandler.h>
+
+#include <cppfs/FileHandle.h>
+
+
+namespace cppfs
+{
+
+
+FileEventHandler::FileEventHandler()
+{
+}
+
+FileEventHandler::~FileEventHandler()
+{
+}
+
+void FileEventHandler::onFileEvent(FileHandle & fh, FileEvent event)
+{
+    if (event & FileCreated) {
```

Comment:
You use bitfield semantics here, whereas the event parameter is constructed using direct assignment instead of bit setters. Do you expect that one event can be of more than one type?
The `FileEvent` enum is also declared using bitfield semantics.

Reply:
I just found code that adds other types by means of bit fields. Seems required.

Reply:
I got confused here (it was very late yesterday ;)). This function will be called with exactly one event type, so we should not use bit field semantic here but just compare the value. When creating the watcher (FileWatcher,add), a bit field combination of event types is accepted, which is why the argument type is unsigned int there instead of FileEvent.